internal/services/healthcare/healthcare_dicom_resource.go

```diff
@@ -239,10 +370,38 @@ func resourceHealthcareApisDicomServiceUpdate(d *pluginsdk.ResourceData, meta in
 		return fmt.Errorf("expanding `identity`: %+v", err)
 	}
 
+	var enableDataPartitions *bool
+	if v, ok := d.GetOk("data_partitions_enabled"); ok {
+		enableDataPartitions = pointer.To(v.(bool))
+	}
+
+	var corsConfiguration *dicomservices.CorsConfiguration
+	if v, ok := d.GetOk("cors"); ok {
+		corsConfiguration = expandDicomServiceCorsConfiguration(v.([]interface{}))
+	}
+
+	var encryption *dicomservices.Encryption
+	if v, ok := d.GetOk("encryption_key_url"); ok {
+		encryption = &dicomservices.Encryption{
+			CustomerManagedKeyEncryption: &dicomservices.EncryptionCustomerManagedKeyEncryption{
+				KeyEncryptionKeyUrl: pointer.To(v.(string)),
+			},
+		}
+	}
+
+	var storageConfiguration *dicomservices.StorageConfiguration
+	if v, ok := d.GetOk("storage"); ok {
+		storageConfiguration = expandStorageConfiguration(v.([]interface{}))
+	}
```

Comment:
We should be using `d.HasChanges` here so we can support the use of `ignore_changes`, you may need to rewrite this to retrieve the existing resource and patch in changes with `d.HasChanges`. The update method in our guide on adding new resources has an explanation on how this should be done:

https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-resource.md

Here is also a practical example from within the provider:
https://github.com/hashicorp/terraform-provider-azurerm/blob/8477e145ee82b79ac723bc86d66f614713d4f0ad/internal/services/network/public_ip_resource.go#L297-L368

Reply:
Got it, have updated the update function in the resource to this format.